core/dbt/contracts/results.py

```diff
@@ -393,6 +393,9 @@ def from_node_results(
         meta = FreshnessMetadata(generated_at=generated_at)
         return cls(metadata=meta, results=results, elapsed_time=elapsed_time)
 
+    def write(self, path):
+        FreshnessExecutionResultArtifact.from_result(self).write(path)
```

Comment:
Yeah we don't stream to any of our output files.  We could look into it, but it would require refactoring and tracking those connections everywhere we use the system client.  Well out of the scope of this ticket IMO.